site/content/en/reference/cfg/search/_index.md

```diff
@@ -0,0 +1,191 @@
+---
+title: "Search"
+linkTitle: "search"
+weight: 4
+type: docs
+description: >
+  Search and optionally replace fields across all resources.
+---
+
+<!--mdtogo:Short
+    Search and optionally replace fields across all resources.
+-->
+
+{{% pageinfo color="primary" %}}
+
+This feature is still under development. Please set environment variable
+`KPT_ENABLE_SEARCH_CMD` to `true` on your local to enable this feature.
+
+{{% /pageinfo %}}
+
+This command leverages the OpenAPI schema to search for fields that may not be
+set on actual configuration files.
+
+Search matchers are provided by flags with `--by-` prefix. When multiple matchers
+are provided they are AND’ed together. `--put-` flags are mutually exclusive.
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+```
+kpt cfg search DIR [flags]
+```
+
+#### Args
+
+```
+DIR:
+  Path to a package directory
+```
+
+<!--mdtogo-->
+
+#### Flags
+
+```sh
+--by-value
+Match by value of a field.
+
+--by-value-regex
+Match by Regex for the value of a field. The syntax of the regular expressions
+accepted is the same general syntax used by Go, Perl, Python, and other languages.
+More precisely, it is the syntax accepted by RE2 and described at
+https://golang.org/s/re2syntax. With the exception that it matches the entire
+value of the field by default without requiring start (^) and end ($) characters.
+
+--by-path
+Match by path expression of a field. See Path Expressions.
+
+--put-literal
+Set or update the value of the matching fields with the given literal value.
+
+--put-pattern
+Put the setter pattern as a line comment for matching fields.
+
+--validate
+Validate the resource configs against openAPI schema.
+
+--recurse-subpackages
+Search recursively in all the nested subpackages.
+```
+
+### Examples
+
+<!--mdtogo:Examples-->
+
+```sh
+# Matches fields with value "3":
+$ kpt cfg search DIR/ --by-value 3
+```
+
+```sh
+# Matches fields with value prefixed by "nginx-":
+$ kpt cfg search DIR/ --by-value-regex ngnix-.*
+```
+
+```sh
+# Matches field with path "spec.namespaces" set to "bookstore":
+$ kpt cfg search DIR/ --by-path metadata.namespace --by-value bookstore
+```
+
+```sh
+# Matches even if namespace is not set on a resource (Using OpenAPI spec):
+$ kpt cfg search DIR/ --by-path metadata.namespace
+```
+
+```sh
+# Matches fields with name "containerPort" arbitrarily deep in "spec" that have value of 80:
+$ kpt cfg search DIR/ --by-path spec.**.containerPort --by-value 80
```

Comment:
These are not all implemented. Are you planning to hide the doc until they are?

Reply:
All the features in the doc are implemented and (few) changes will be merged before the fishfood. This doc can be reviewed during fishfood.